test/authority_test.go

```diff
@@ -80,15 +80,16 @@ func runUnixTest(t *testing.T, address, target, expectedAuthority string, dialer
 	}
 }
 
-// TestUnix does end to end tests with the various supported unix target
-// formats, ensuring that the authority is set to localhost in every case.
-func (s) TestUnix(t *testing.T) {
-	tests := []struct {
-		name      string
-		address   string
-		target    string
-		authority string
-	}{
+type authorityTest struct {
+	name           string
+	address        string
+	target         string
+	authority      string
+	expectedTarget string
+}
+
+func tests() []authorityTest {
```

Comment:
Use a variable not a function.  Also, need a better name than `tests` since this is package-global.

OR, make this non-global and put it in `TestUnix`, with a `runUnixCustomDialerTest` or something for the other tests.

Reply:
Made it a variable named `authorityTests`